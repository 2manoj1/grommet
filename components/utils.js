'use strict';

exports.__esModule = true;
exports.focusStyle = undefined;
exports.parseMetricToInt = parseMetricToInt;
exports.findAllByType = findAllByType;

var _styledComponents = require('styled-components');

var focusStyle = exports.focusStyle = (0, _styledComponents.css)(['border-color:', ';box-shadow:0 0 1px 1px ', ';'], function (props) {
  return props.theme.global.focus.border.color || props.theme.global.colors.accent[0];
}, function (props) {
  return props.theme.global.focus.border.color || props.theme.global.colors.accent[0];
});

function parseMetricToInt(fontAsString) {
  return parseInt(fontAsString.replace(/[^0-9]/g, ''), 10);
}

function findAllByType(component, type) {
  var matches = [];

  if (component.type === type) {
    matches.push(component);
  }

  if (component.children) {
    component.children.forEach(function (child) {
      matches = matches.concat(findAllByType(child, type));
    });
  }

  return matches;
}

exports.default = { findAllByType: findAllByType, focusStyle: focusStyle, parseMetricToInt: parseMetricToInt };