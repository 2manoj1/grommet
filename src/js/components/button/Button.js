import React, { Children, Component } from 'react';
import PropTypes from 'prop-types';
import deepAssign from 'deep-assign';

import StyledButton, { StyledLabel, StyledIcon } from './StyledButton';

import { withFocus } from '../hocs';

import doc from './doc';

class Button extends Component {
  static contextTypes = {
    theme: PropTypes.object.isRequired,
  }
  render() {
    const {
      a11yTitle,
      box,
      children,
      icon,
      focus,
      href,
      label,
      onClick,
      plain,
      reverse,
      theme,
      type,
      ...rest
    } = this.props;
    const { theme: contextTheme } = this.context;
    const localTheme = deepAssign(contextTheme, theme);

    let Tag = StyledButton;
    if (href) {
      Tag = Tag.withComponent('a');
    }

    let boxProps;
    if (box) {
      // Let the root element of the Button be a Box element with tag prop
      boxProps = {
        tag: href ? 'a' : 'button',
      };
      // TODO: replace it with Box
      Tag = Tag.withComponent('div');
    }

    let buttonIcon;
    if (icon) {
      buttonIcon = <StyledIcon theme={localTheme}>{icon}</StyledIcon>;
    }

    let buttonLabel;
    if (label) {
      buttonLabel = <StyledLabel theme={localTheme}>{label}</StyledLabel>;
    }

    const first = reverse ? buttonLabel : buttonIcon;
    const second = reverse ? buttonIcon : buttonLabel;

    const disabled = (
      !href &&
      !onClick &&
      ['reset', 'submit'].indexOf(type) === -1
    );

    const plainProp = (
      plain ||
      box ||
      Children.count(children) > 0 ||
      (icon && !label)
    );

    return (
      <Tag
        {...rest}
        {...boxProps}
        aria-label={a11yTitle}
        box={box}
        disabled={disabled}
        focus={focus}
        href={href}
        onClick={onClick}
        plain={plainProp}
        theme={localTheme}
        type={type}
      >
        {first}
        {second}
        {children}
      </Tag>
    );
  }
}

if (process.env.NODE_ENV !== 'production') {
  doc(Button);
}

export default withFocus(Button);
