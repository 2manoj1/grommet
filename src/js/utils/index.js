export * from './background';
export * from './colors';
export * from './debounce';
export * from './DOM';
export * from './graphics';
export * from './mixins';
export * from './object';
export * from './prop-types';
export * from './responsive';
export * from './router';
export * from './styles';
export * from './media';
export * from './throttle';
export * from './themeDocUtils';
