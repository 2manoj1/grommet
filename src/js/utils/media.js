// return the time in a format of minutes:seconds. if time is not defined, return '' to avoid NaN.
export const formatTime = time => {
  if (time) {
    let minutes = Math.round(time / 60);
    if (minutes < 10) {
      minutes = `0${minutes}`;
    }
    let seconds = Math.round(time) % 60;
    if (seconds < 10) {
      seconds = `0${seconds}`;
    }
    return `${minutes}:${seconds}`;
  }
  return '';
};

export const mediaEvents = [
  'onAbort',
  'onCanPlay',
  'onCanPlayThrough',
  'onDurationChange',
  'onEmptied',
  'onEncrypted',
  'onEnded',
  'onError',
  'onLoadedData',
  'onLoadedMetadata',
  'onLoadStart',
  'onPause',
  'onPlay',
  'onPlaying',
  'onProgress',
  'onRateChange',
  'onSeeked',
  'onSeeking',
  'onStalled',
  'onSuspend',
  'onTimeUpdate',
  'onVolumeChange',
  'onWaiting',
];
