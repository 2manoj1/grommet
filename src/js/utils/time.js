export const formatTime = time => {
    let minutes = Math.round(time / 60);
    if (minutes < 10) {
        minutes = `0${minutes}`;
    }
    let seconds = Math.round(time) % 60;
    if (seconds < 10) {
        seconds = `0${seconds}`;
    }
    return `${minutes}:${seconds}`;
};