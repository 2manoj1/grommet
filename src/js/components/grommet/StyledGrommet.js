import styled, { css } from 'styled-components';

import { lapAndUp } from '../mixins';
import { parseMetricToInt } from '../utils';

const centeredStyle = css`
  width: 100%;
  max-width: ${props => props.theme.brand.centerColumnWidth};
  margin-left: auto;
  margin-right: auto;
`;

const inlineStyle = css`
  position: relative;
`;

const StyledGrommet = styled.div`
  font-family: ${props => props.theme.brand.font.family};
  font-size: ${props => `${(parseMetricToInt(props.theme.brand.font.size) / 16) * 1}em`};
  line-height: ${props => (
    parseMetricToInt(props.theme.brand.lineHeight) / parseMetricToInt(props.theme.brand.font.size)
  )};
  color: ${props => props.theme.colors.text};
  background-color: ${props => props.theme.colors.background};

  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;

  * {
    box-sizing: inherit;
  }

  ${lapAndUp(`
    top: 0px;
    bottom: 0px;
    left: 0px;
    right: 0px;
    height: 100%;
    width: 100%;
    overflow: visible;
  `)}

  ${props => props.centered && centeredStyle}
  ${props => props.inline && inlineStyle}
  ${props => props.theme.brand.font.face}
  ${props => props.theme.grommet.extend}
`;

export default StyledGrommet;
