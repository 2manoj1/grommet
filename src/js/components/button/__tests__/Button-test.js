import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Grommet from '../../grommet';
import Button from '../';

import { findAllByType } from '../../utils';

test('Button renders', () => {
  const component = renderer.create(
    <Grommet>
      <Button label='Test' onClick={() => {}} />
    </Grommet>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Button primary renders', () => {
  const component = renderer.create(
    <Grommet>
      <Button primary={true} label='Test' onClick={() => {}} />
    </Grommet>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Button accent renders', () => {
  const component = renderer.create(
    <Grommet>
      <Button accent={true} label='Test' onClick={() => {}} />
    </Grommet>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Button secondary renders', () => {
  const component = renderer.create(
    <Grommet>
      <Button secondary={true} label='Test' onClick={() => {}} />
    </Grommet>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Button critical renders', () => {
  const component = renderer.create(
    <Grommet>
      <Button critical={true} label='Test' onClick={() => {}} />
    </Grommet>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Button focus renders', () => {
  const component = renderer.create(
    <Grommet>
      <Button focus={true} label='Test' onClick={() => {}} />
    </Grommet>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Button disabled renders', () => {
  const component = renderer.create(
    <Grommet>
      <Button />
    </Grommet>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Button icon label renders', () => {
  const component = renderer.create(
    <Grommet>
      <Button icon={<svg />} label='Test' onClick={() => {}} />
    </Grommet>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Button reverse icon label renders', () => {
  const component = renderer.create(
    <Grommet>
      <Button
        reverse={true}
        icon={<svg />}
        label='Test'
        onClick={() => {}}
      />
    </Grommet>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Button href renders', () => {
  const component = renderer.create(
    <Grommet>
      <Button href='test' />
    </Grommet>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Button box renders', () => {
  const component = renderer.create(
    <Grommet>
      <Button box={true} label='Test' onClick={() => {}} />
    </Grommet>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Button box href renders', () => {
  const component = renderer.create(
    <Grommet>
      <Button box={true} href='test' />
    </Grommet>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Button is clickable', () => {
  const onClick = jest.fn();
  const component = renderer.create(
    <Grommet>
      <Button label='Test' onClick={onClick} />
    </Grommet>
  );
  const tree = component.toJSON();

  const button = findAllByType(tree, 'button');
  button[0].props.onClick();
  expect(onClick).toBeCalled();
});
