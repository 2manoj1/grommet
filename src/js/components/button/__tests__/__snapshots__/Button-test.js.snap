// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button accent renders 1`] = `
.c0 {
  font-family: 'Work Sans', Arial, sans-serif;
  font-size: 1em;
  line-height: 1.5;
  color: #333333;
  background-color: #FFFFFF;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  width: 100%;
  max-width: 1152px;
  margin-left: auto;
  margin-right: auto;
}

.c0 * {
  box-sizing: inherit;
}

.c2:first-child:not(:last-child) {
  margin-right: 12px;
}

.c1 {
  border: 2px solid #865CD6;
  border-radius: 5px;
  color: #333333;
  cursor: pointer;
  text-align: center;
  outline: none;
  min-width: 120px;
  max-width: 384px;
  margin: 0;
  overflow: visible;
  text-transform: none;
  background-color: transparent;
  -webkit-appearance: button;
  font: inherit;
  font-weight: 600;
  font-size: 1.1875rem;
  line-height: 24px;
  padding: 6px 22px;
  border-color: #00CCEB;
}

.c1:hover {
  box-shadow: 0px 0px 0px 2px #00CCEB;
}

.c1:hover svg {
  fill: #000000;
  stroke: #000000;
  -webkit-transition: none;
  transition: none;
}

@media only screen and (min-width:481px) {
  .c0 {
    top: 0px;
    bottom: 0px;
    left: 0px;
    right: 0px;
    height: 100%;
    width: 100%;
    overflow: visible;
  }
}

@media only screen and (min-width:481px) {
  .c1 {
    -webkit-transition: 0.1s ease-in-out;
    transition: 0.1s ease-in-out;
  }
}

<div
  className="c0"
>
  <button
    aria-label={undefined}
    className="c1"
    disabled={false}
    href={undefined}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    type="button"
  >
    <span
      className="c2"
    >
      Test
    </span>
  </button>
</div>
`;

exports[`Button box href renders 1`] = `
.c0 {
  font-family: 'Work Sans', Arial, sans-serif;
  font-size: 1em;
  line-height: 1.5;
  color: #333333;
  background-color: #FFFFFF;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  width: 100%;
  max-width: 1152px;
  margin-left: auto;
  margin-right: auto;
}

.c0 * {
  box-sizing: inherit;
}

.c1 {
  border: 2px solid #865CD6;
  border-radius: 5px;
  color: #333333;
  cursor: pointer;
  text-align: center;
  outline: none;
  min-width: 120px;
  max-width: 384px;
  margin: 0;
  overflow: visible;
  text-transform: none;
  background-color: transparent;
  -webkit-appearance: button;
  font: inherit;
  font-weight: 600;
  font-size: 1.1875rem;
  line-height: 24px;
  width: auto;
  height: auto;
  min-width: 0;
  max-width: none;
  text-align: inherit;
  font-weight: inherit;
  padding: 6px;
  border-color: transparent;
}

@media only screen and (min-width:481px) {
  .c0 {
    top: 0px;
    bottom: 0px;
    left: 0px;
    right: 0px;
    height: 100%;
    width: 100%;
    overflow: visible;
  }
}

@media only screen and (min-width:481px) {
  .c1 {
    -webkit-transition: 0.1s ease-in-out;
    transition: 0.1s ease-in-out;
  }
}

<div
  className="c0"
>
  <div
    aria-label={undefined}
    className="c1"
    disabled={false}
    href="test"
    onBlur={[Function]}
    onClick={undefined}
    onFocus={[Function]}
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    type="button"
  />
</div>
`;

exports[`Button box renders 1`] = `
.c0 {
  font-family: 'Work Sans', Arial, sans-serif;
  font-size: 1em;
  line-height: 1.5;
  color: #333333;
  background-color: #FFFFFF;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  width: 100%;
  max-width: 1152px;
  margin-left: auto;
  margin-right: auto;
}

.c0 * {
  box-sizing: inherit;
}

.c2:first-child:not(:last-child) {
  margin-right: 12px;
}

.c1 {
  border: 2px solid #865CD6;
  border-radius: 5px;
  color: #333333;
  cursor: pointer;
  text-align: center;
  outline: none;
  min-width: 120px;
  max-width: 384px;
  margin: 0;
  overflow: visible;
  text-transform: none;
  background-color: transparent;
  -webkit-appearance: button;
  font: inherit;
  font-weight: 600;
  font-size: 1.1875rem;
  line-height: 24px;
  width: auto;
  height: auto;
  min-width: 0;
  max-width: none;
  text-align: inherit;
  font-weight: inherit;
  padding: 6px;
  border-color: transparent;
}

@media only screen and (min-width:481px) {
  .c0 {
    top: 0px;
    bottom: 0px;
    left: 0px;
    right: 0px;
    height: 100%;
    width: 100%;
    overflow: visible;
  }
}

@media only screen and (min-width:481px) {
  .c1 {
    -webkit-transition: 0.1s ease-in-out;
    transition: 0.1s ease-in-out;
  }
}

<div
  className="c0"
>
  <div
    aria-label={undefined}
    className="c1"
    disabled={false}
    href={undefined}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    type="button"
  >
    <span
      className="c2"
    >
      Test
    </span>
  </div>
</div>
`;

exports[`Button critical renders 1`] = `
.c0 {
  font-family: 'Work Sans', Arial, sans-serif;
  font-size: 1em;
  line-height: 1.5;
  color: #333333;
  background-color: #FFFFFF;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  width: 100%;
  max-width: 1152px;
  margin-left: auto;
  margin-right: auto;
}

.c0 * {
  box-sizing: inherit;
}

.c2:first-child:not(:last-child) {
  margin-right: 12px;
}

.c1 {
  border: 2px solid #865CD6;
  border-radius: 5px;
  color: #333333;
  cursor: pointer;
  text-align: center;
  outline: none;
  min-width: 120px;
  max-width: 384px;
  margin: 0;
  overflow: visible;
  text-transform: none;
  background-color: transparent;
  -webkit-appearance: button;
  font: inherit;
  font-weight: 600;
  font-size: 1.1875rem;
  line-height: 24px;
  padding: 6px 22px;
  border-color: #FF324D;
}

.c1:hover {
  box-shadow: 0px 0px 0px 2px #FF324D;
}

.c1:hover svg {
  fill: #000000;
  stroke: #000000;
  -webkit-transition: none;
  transition: none;
}

@media only screen and (min-width:481px) {
  .c0 {
    top: 0px;
    bottom: 0px;
    left: 0px;
    right: 0px;
    height: 100%;
    width: 100%;
    overflow: visible;
  }
}

@media only screen and (min-width:481px) {
  .c1 {
    -webkit-transition: 0.1s ease-in-out;
    transition: 0.1s ease-in-out;
  }
}

<div
  className="c0"
>
  <button
    aria-label={undefined}
    className="c1"
    disabled={false}
    href={undefined}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    type="button"
  >
    <span
      className="c2"
    >
      Test
    </span>
  </button>
</div>
`;

exports[`Button disabled renders 1`] = `
.c0 {
  font-family: 'Work Sans', Arial, sans-serif;
  font-size: 1em;
  line-height: 1.5;
  color: #333333;
  background-color: #FFFFFF;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  width: 100%;
  max-width: 1152px;
  margin-left: auto;
  margin-right: auto;
}

.c0 * {
  box-sizing: inherit;
}

.c1 {
  border: 2px solid #865CD6;
  border-radius: 5px;
  color: #333333;
  cursor: pointer;
  text-align: center;
  outline: none;
  min-width: 120px;
  max-width: 384px;
  margin: 0;
  overflow: visible;
  text-transform: none;
  background-color: transparent;
  -webkit-appearance: button;
  font: inherit;
  font-weight: 600;
  opacity: 0.3;
  cursor: default;
  font-size: 1.1875rem;
  line-height: 24px;
  padding: 6px 22px;
}

@media only screen and (min-width:481px) {
  .c0 {
    top: 0px;
    bottom: 0px;
    left: 0px;
    right: 0px;
    height: 100%;
    width: 100%;
    overflow: visible;
  }
}

@media only screen and (min-width:481px) {
  .c1 {
    -webkit-transition: 0.1s ease-in-out;
    transition: 0.1s ease-in-out;
  }
}

<div
  className="c0"
>
  <button
    aria-label={undefined}
    className="c1"
    disabled={true}
    href={undefined}
    onBlur={[Function]}
    onClick={undefined}
    onFocus={[Function]}
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    type="button"
  />
</div>
`;

exports[`Button focus renders 1`] = `
.c0 {
  font-family: 'Work Sans', Arial, sans-serif;
  font-size: 1em;
  line-height: 1.5;
  color: #333333;
  background-color: #FFFFFF;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  width: 100%;
  max-width: 1152px;
  margin-left: auto;
  margin-right: auto;
}

.c0 * {
  box-sizing: inherit;
}

.c2:first-child:not(:last-child) {
  margin-right: 12px;
}

.c1 {
  border: 2px solid #865CD6;
  border-radius: 5px;
  color: #333333;
  cursor: pointer;
  text-align: center;
  outline: none;
  min-width: 120px;
  max-width: 384px;
  margin: 0;
  overflow: visible;
  text-transform: none;
  background-color: transparent;
  -webkit-appearance: button;
  font: inherit;
  font-weight: 600;
  font-size: 1.1875rem;
  line-height: 24px;
  padding: 6px 22px;
  border-color: #00CCEB;
  box-shadow: 0 0 1px 1px #00CCEB;
}

@media only screen and (min-width:481px) {
  .c0 {
    top: 0px;
    bottom: 0px;
    left: 0px;
    right: 0px;
    height: 100%;
    width: 100%;
    overflow: visible;
  }
}

@media only screen and (min-width:481px) {
  .c1 {
    -webkit-transition: 0.1s ease-in-out;
    transition: 0.1s ease-in-out;
  }
}

<div
  className="c0"
>
  <button
    aria-label={undefined}
    className="c1"
    disabled={false}
    href={undefined}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    type="button"
  >
    <span
      className="c2"
    >
      Test
    </span>
  </button>
</div>
`;

exports[`Button href renders 1`] = `
.c0 {
  font-family: 'Work Sans', Arial, sans-serif;
  font-size: 1em;
  line-height: 1.5;
  color: #333333;
  background-color: #FFFFFF;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  width: 100%;
  max-width: 1152px;
  margin-left: auto;
  margin-right: auto;
}

.c0 * {
  box-sizing: inherit;
}

.c1 {
  border: 2px solid #865CD6;
  border-radius: 5px;
  color: #333333;
  cursor: pointer;
  text-align: center;
  outline: none;
  min-width: 120px;
  max-width: 384px;
  margin: 0;
  overflow: visible;
  text-transform: none;
  background-color: transparent;
  -webkit-appearance: button;
  font: inherit;
  font-weight: 600;
  font-size: 1.1875rem;
  line-height: 24px;
  padding: 6px 22px;
}

.c1:hover {
  box-shadow: 0px 0px 0px 2px #865CD6;
}

.c1:hover svg {
  fill: #000000;
  stroke: #000000;
  -webkit-transition: none;
  transition: none;
}

@media only screen and (min-width:481px) {
  .c0 {
    top: 0px;
    bottom: 0px;
    left: 0px;
    right: 0px;
    height: 100%;
    width: 100%;
    overflow: visible;
  }
}

@media only screen and (min-width:481px) {
  .c1 {
    -webkit-transition: 0.1s ease-in-out;
    transition: 0.1s ease-in-out;
  }
}

<div
  className="c0"
>
  <a
    aria-label={undefined}
    className="c1"
    disabled={false}
    href="test"
    onBlur={[Function]}
    onClick={undefined}
    onFocus={[Function]}
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    type="button"
  />
</div>
`;

exports[`Button icon label renders 1`] = `
.c0 {
  font-family: 'Work Sans', Arial, sans-serif;
  font-size: 1em;
  line-height: 1.5;
  color: #333333;
  background-color: #FFFFFF;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  width: 100%;
  max-width: 1152px;
  margin-left: auto;
  margin-right: auto;
}

.c0 * {
  box-sizing: inherit;
}

.c3:first-child:not(:last-child) {
  margin-right: 12px;
}

.c2 {
  display: inline-block;
}

.c2:first-child:not(:last-child) {
  margin-right: 12px;
}

.c2 > * {
  vertical-align: bottom;
}

.c1 {
  border: 2px solid #865CD6;
  border-radius: 5px;
  color: #333333;
  cursor: pointer;
  text-align: center;
  outline: none;
  min-width: 120px;
  max-width: 384px;
  margin: 0;
  overflow: visible;
  text-transform: none;
  background-color: transparent;
  -webkit-appearance: button;
  font: inherit;
  font-weight: 600;
  font-size: 1.1875rem;
  line-height: 24px;
  padding: 6px 22px;
}

.c1:hover {
  box-shadow: 0px 0px 0px 2px #865CD6;
}

.c1:hover svg {
  fill: #000000;
  stroke: #000000;
  -webkit-transition: none;
  transition: none;
}

@media only screen and (min-width:481px) {
  .c0 {
    top: 0px;
    bottom: 0px;
    left: 0px;
    right: 0px;
    height: 100%;
    width: 100%;
    overflow: visible;
  }
}

@media only screen and (min-width:481px) {
  .c1 {
    -webkit-transition: 0.1s ease-in-out;
    transition: 0.1s ease-in-out;
  }
}

<div
  className="c0"
>
  <button
    aria-label={undefined}
    className="c1"
    disabled={false}
    href={undefined}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    type="button"
  >
    <span
      className="c2"
    >
      <svg />
    </span>
    <span
      className="c3"
    >
      Test
    </span>
  </button>
</div>
`;

exports[`Button primary renders 1`] = `
.c0 {
  font-family: 'Work Sans', Arial, sans-serif;
  font-size: 1em;
  line-height: 1.5;
  color: #333333;
  background-color: #FFFFFF;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  width: 100%;
  max-width: 1152px;
  margin-left: auto;
  margin-right: auto;
}

.c0 * {
  box-sizing: inherit;
}

.c2:first-child:not(:last-child) {
  margin-right: 12px;
}

.c1 {
  border: 2px solid #865CD6;
  border-radius: 5px;
  color: #333333;
  cursor: pointer;
  text-align: center;
  outline: none;
  min-width: 120px;
  max-width: 384px;
  margin: 0;
  overflow: visible;
  text-transform: none;
  background-color: transparent;
  -webkit-appearance: button;
  font: inherit;
  font-weight: 600;
  font-size: 1.1875rem;
  line-height: 24px;
  padding: 6px 22px;
  background-color: #865CD6;
  color: #FFFFFF;
}

.c1:hover {
  box-shadow: 0px 0px 0px 2px #865CD6;
}

.c1 svg {
  fill: #FFFFFF;
  stroke: #FFFFFF;
  -webkit-transition: none;
  transition: none;
}

@media only screen and (min-width:481px) {
  .c0 {
    top: 0px;
    bottom: 0px;
    left: 0px;
    right: 0px;
    height: 100%;
    width: 100%;
    overflow: visible;
  }
}

@media only screen and (min-width:481px) {
  .c1 {
    -webkit-transition: 0.1s ease-in-out;
    transition: 0.1s ease-in-out;
  }
}

<div
  className="c0"
>
  <button
    aria-label={undefined}
    className="c1"
    disabled={false}
    href={undefined}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    type="button"
  >
    <span
      className="c2"
    >
      Test
    </span>
  </button>
</div>
`;

exports[`Button renders 1`] = `
.c0 {
  font-family: 'Work Sans', Arial, sans-serif;
  font-size: 1em;
  line-height: 1.5;
  color: #333333;
  background-color: #FFFFFF;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  width: 100%;
  max-width: 1152px;
  margin-left: auto;
  margin-right: auto;
}

.c0 * {
  box-sizing: inherit;
}

.c2:first-child:not(:last-child) {
  margin-right: 12px;
}

.c1 {
  border: 2px solid #865CD6;
  border-radius: 5px;
  color: #333333;
  cursor: pointer;
  text-align: center;
  outline: none;
  min-width: 120px;
  max-width: 384px;
  margin: 0;
  overflow: visible;
  text-transform: none;
  background-color: transparent;
  -webkit-appearance: button;
  font: inherit;
  font-weight: 600;
  font-size: 1.1875rem;
  line-height: 24px;
  padding: 6px 22px;
}

.c1:hover {
  box-shadow: 0px 0px 0px 2px #865CD6;
}

.c1:hover svg {
  fill: #000000;
  stroke: #000000;
  -webkit-transition: none;
  transition: none;
}

@media only screen and (min-width:481px) {
  .c0 {
    top: 0px;
    bottom: 0px;
    left: 0px;
    right: 0px;
    height: 100%;
    width: 100%;
    overflow: visible;
  }
}

@media only screen and (min-width:481px) {
  .c1 {
    -webkit-transition: 0.1s ease-in-out;
    transition: 0.1s ease-in-out;
  }
}

<div
  className="c0"
>
  <button
    aria-label={undefined}
    className="c1"
    disabled={false}
    href={undefined}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    type="button"
  >
    <span
      className="c2"
    >
      Test
    </span>
  </button>
</div>
`;

exports[`Button reverse icon label renders 1`] = `
.c0 {
  font-family: 'Work Sans', Arial, sans-serif;
  font-size: 1em;
  line-height: 1.5;
  color: #333333;
  background-color: #FFFFFF;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  width: 100%;
  max-width: 1152px;
  margin-left: auto;
  margin-right: auto;
}

.c0 * {
  box-sizing: inherit;
}

.c2:first-child:not(:last-child) {
  margin-right: 12px;
}

.c3 {
  display: inline-block;
}

.c3:first-child:not(:last-child) {
  margin-right: 12px;
}

.c3 > * {
  vertical-align: bottom;
}

.c1 {
  border: 2px solid #865CD6;
  border-radius: 5px;
  color: #333333;
  cursor: pointer;
  text-align: center;
  outline: none;
  min-width: 120px;
  max-width: 384px;
  margin: 0;
  overflow: visible;
  text-transform: none;
  background-color: transparent;
  -webkit-appearance: button;
  font: inherit;
  font-weight: 600;
  font-size: 1.1875rem;
  line-height: 24px;
  padding: 6px 22px;
}

.c1:hover {
  box-shadow: 0px 0px 0px 2px #865CD6;
}

.c1:hover svg {
  fill: #000000;
  stroke: #000000;
  -webkit-transition: none;
  transition: none;
}

@media only screen and (min-width:481px) {
  .c0 {
    top: 0px;
    bottom: 0px;
    left: 0px;
    right: 0px;
    height: 100%;
    width: 100%;
    overflow: visible;
  }
}

@media only screen and (min-width:481px) {
  .c1 {
    -webkit-transition: 0.1s ease-in-out;
    transition: 0.1s ease-in-out;
  }
}

<div
  className="c0"
>
  <button
    aria-label={undefined}
    className="c1"
    disabled={false}
    href={undefined}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    type="button"
  >
    <span
      className="c2"
    >
      Test
    </span>
    <span
      className="c3"
    >
      <svg />
    </span>
  </button>
</div>
`;

exports[`Button secondary renders 1`] = `
.c0 {
  font-family: 'Work Sans', Arial, sans-serif;
  font-size: 1em;
  line-height: 1.5;
  color: #333333;
  background-color: #FFFFFF;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  width: 100%;
  max-width: 1152px;
  margin-left: auto;
  margin-right: auto;
}

.c0 * {
  box-sizing: inherit;
}

.c2:first-child:not(:last-child) {
  margin-right: 12px;
}

.c1 {
  border: 2px solid #865CD6;
  border-radius: 5px;
  color: #333333;
  cursor: pointer;
  text-align: center;
  outline: none;
  min-width: 120px;
  max-width: 384px;
  margin: 0;
  overflow: visible;
  text-transform: none;
  background-color: transparent;
  -webkit-appearance: button;
  font: inherit;
  font-weight: 600;
  font-size: 1.1875rem;
  line-height: 24px;
  padding: 6px 22px;
  border-color: #DC2878;
}

.c1:hover {
  box-shadow: 0px 0px 0px 2px #DC2878;
}

.c1:hover svg {
  fill: #000000;
  stroke: #000000;
  -webkit-transition: none;
  transition: none;
}

@media only screen and (min-width:481px) {
  .c0 {
    top: 0px;
    bottom: 0px;
    left: 0px;
    right: 0px;
    height: 100%;
    width: 100%;
    overflow: visible;
  }
}

@media only screen and (min-width:481px) {
  .c1 {
    -webkit-transition: 0.1s ease-in-out;
    transition: 0.1s ease-in-out;
  }
}

<div
  className="c0"
>
  <button
    aria-label={undefined}
    className="c1"
    disabled={false}
    href={undefined}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    type="button"
  >
    <span
      className="c2"
    >
      Test
    </span>
  </button>
</div>
`;
